### ISSUE TITLE
Reconnection timeout on VPN

### ISSUE THREAD

**Issue by simmkyu** [aws/amazon-chime-sdk-js#1985]
### What happened and what did you expect to happen?

The Chime SDK for JavaScript reconnection timed out when turning WiFI on/off on the VPN connection.

### Have you reviewed our existing documentation?

- [X] GitHub [issues](https://github.com/aws/amazon-chime-sdk-js/issues)
- [X] FAQs [page](https://aws.github.io/amazon-chime-sdk-js/modules/faqs.html)
- [X] README [page](https://github.com/aws/amazon-chime-sdk-js#amazon-chime-sdk-for-javascript)
- [X] Feature-specific [guides](https://github.com/aws/amazon-chime-sdk-js#guides)

### Reproduction steps

1. Connect to VPN.
1. Join a meeting.
1. Turn your WiFi off.
1.  Wait several seconds.
1. Turn your WiFi on and wait for your VPN to reconnect.
1. SDK attempts to retry the connection.

### Amazon Chime SDK for JavaScript version

2.26.0

### What browsers are you seeing the problem on?

Chrome and Chromium-based environments like Electron

### Browser version

Chrome 97.0.4692.99

### Meeting and Attendee ID Information.

Not required

## Timeout

SDK does not publish any event after `WebSocketClosing`.

```
00:00:00.097 adding connection request to queue: wss://__chime_signaling_server_url__
00:00:00.097 notifying event: WebSocketClosing
00:00:15.087 timeout reached for task AudioVideoReconnect/__your_meeting_id__/__your_attendee_id__/Timeout15000ms
```

## Success

SDK should publish `WebSocketConnecting` and `WebSocketOpen` events immediately after `WebSocketClosing`.

```
00:00:00.504Z adding connection request to queue: wss://__chime_signaling_server_url__
00:00:00.504Z notifying event: WebSocketClosing
00:00:00.752Z opening connection to wss://__chime_signaling_server_url__
00:00:00.753Z notifying event: WebSocketConnecting
00:00:00.898Z notifying event: WebSocketOpen
00:00:00.898Z Opening signaling connection took 393 ms
```

**Comment by simmkyu** [comment]
## Root cause

Suppose that VPN is enabled.  The root cause is that an existing WebSocket object does not receive the `close` event when calling the `WebSocket.close` API. The Chime SDK for JavaScript opens a new connection only when receiving the `WebSocket.close` event during retries.

## Reproducible steps without using the Chime SDK for JavaScript

The bug is reproducible in the WebSocket-based app without using the Chime SDK for JavaScript. The bug only occurs in Chrome (Tested on 97.0.4692.99) and Chromium environments like Electron.

1. Deploy the [API Gateway WebSocket example](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:729047367331:applications~simple-websockets-chat-app) to your AWS account.
1. Connect to VPN.
1. Replace `{ }` with your API Gateway output and run the following script in the browser console.
   ```
    var ws = new WebSocket('wss://{YOUR-API-ID}.execute-api.{YOUR-REGION}.amazonaws.com/{STAGE}');

    ws.addEventListener('open', () => {
      console.warn('connected!');

      const interval = setInterval(() => {
        ws.send(JSON.stringify({"action":"sendmessage", "data":"hello world"}));
      }, 2000);
      ws.addEventListener('message', event => {
        console.warn('received: ', event);
      });
      ws.addEventListener('close', event => {
        console.warn('closed: ', event);
        clearInterval(interval);
      });
      ws.addEventListener('error', event => {
        console.warn('error: ', event);
        clearInterval(interval);
      });
    });
   ```
1. You should see the `received` message every two seconds in the browser console.
1. Turn your WiFi off. Ensure that you stop seeing the `received` message.
1. After 10 seconds, turn your WiFi on. You should not see any `received` message.
1. Type and press enter. WebSocket is still open.
    ```
    ws.readyState // 1 (OPEN)
    ```
1. Type and press enter. **You do not see the `closed` message.**
    ```
    ws.close();
    ```
1. Type and press enter. WebSocket gets stuck in the closing state.
    ```
    ws.readyState // 2 (CLOSING)
    ```

**Comment by vidya-mohan** [comment]
This is released in 2.27
